docs/embedding.md

```diff
@@ -0,0 +1,106 @@
+
+> Remember, a 70B model is going to be __almost 130GB__ in size, if you're looking to download that using a simple curl command, you're going to be waiting for a very long time. 
+
+We could use something like S3 but that means that for large datasets and weights, we'd be eating significant network eggress charges while simultaneously dealing with slow transfer speeds and rate limits.
+
+That's where Volumes come in - they allow us to have our target data right beside our GPU and load it into our containers concurrently. The feature itself is built to allow for all of your containers to read in data quickly and efficiently at the same time. This is ideal for jobs where we might need to run a large batch job involving a distributed dataset or a model whose weights we'd like to load in quickly. 
+
+It's suprisingly simple to set-up, all you need to create a new volume is the following 5 lines of code
+
+```python
+import modal
+
+stub = modal.Stub()
+stub.volume = modal.volume.new() # Declare a new volume
+
+@stub.function(volumes={"/root/foo": stub.volume})
+def function():
+  // Perform tasks here
+```
+
+If you'd like to ensure that the data in the volume is permanently stored on Modal's servers and can be accessed by any of your other modal functions, all it takes is just swapping out the `.new` for the `.persisted` keyword
+
+```python
+volume = Volume.persisted("Unique Name Here")
+```
+
+Once you have this, you'll be able to access this volume from any other app you define using the `volumes` parameter in the `stub.function` keyword. In fact, what makes this so good is that a file on your volume behaves almost identically to a file located on your local file system, yet mantains all of the advantages that Modal brings. 
+
+In some of our initial test runs, we were able to load in the entirety of English Wikipedia, all 22 GB of it, into a container in just under 5 seconds and start executing jobs. Compare that to the 10 minutes it took to download the dataset from scratch and you can imagine the huge benefits this would bring for any endpoints or batch jobs down the line. 
+
+
+
+# Embedding Wikipedia
+
+Now that we've got a good understanding of some key concepts that Modal provides, let's look at volumes in action by running our first embedding job. We've loaded a small dataset into a directory called `wikipedia` inside a persistent volume we've created called `embedding-wikipedia`
+
+<TODO : Add in a simplified script to run embeddings without the sample portion>
+
+## Scaling Out
+
+Now that we've seen how to run a simple batch job using Modal, let's consider how we might modify our embedding function to bring our current timing of ~4 hours down. 
+
+Well, turns out we have two handy properties that we can adjust
+
+- `concurrency_limit` : This is the number of containers that Modal is allowed to spawn concurrently to run a batch job
```

Comment:
For some reason, modal lists the default values of these two as None. Not too sure what we can assume how the paid tier works.